test/unittest/dds/xtypes/type_representation/TypeObjectUtilsTests.cpp

```diff
@@ -461,93 +495,398 @@ TEST(TypeObjectUtilsTests, build_plain_map_s_type_defn_inconsistencies)
+        minimal_typeid = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier2()));
     }
     else
     {
-        test_identifier = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier2()));
+        complete_typeid = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier2()));
+        minimal_typeid = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier1()));
     }
-    // TypeIdentifier consistent with complete header
-    EXPECT_NO_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
-                complete_header, 10, test_identifier, flags, key_identifier));
-    // TypeIdentifier inconsistent with minimal header
-    EXPECT_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
-                minimal_header, 10, test_identifier, flags, key_identifier), InvalidArgumentError);
-    // TypeIdentifier inconsistent with fully-descriptive header
-    EXPECT_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
-                fully_descriptive_header, 10, test_identifier, flags, key_identifier), InvalidArgumentError);
-    // Change discriminator to EK_MINIMAL
-    if (EK_COMPLETE == type_ids.type_identifier1()._d())
+
+    register_plain_seq_type_object(complete_typeid);
+    TypeIdentifierPair indirect_type_ids;
+    eprosima::fastcdr::external<TypeIdentifier> complete_indirect_id;
+    eprosima::fastcdr::external<TypeIdentifier> minimal_indirect_id;
+    DomainParticipantFactory::get_instance()->type_object_registry().get_type_identifiers("plain_sequence",
+            indirect_type_ids);
+    if (EK_COMPLETE == indirect_type_ids.type_identifier1()._d())
     {
-        test_identifier = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier2()));
+        complete_indirect_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(indirect_type_ids.type_identifier1()));
+        minimal_indirect_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(indirect_type_ids.type_identifier2()));
     }
     else
     {
-        test_identifier = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier1()));
+        complete_indirect_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(indirect_type_ids.type_identifier2()));
+        minimal_indirect_id =
+                eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(indirect_type_ids.type_identifier1()));
     }
-    // TypeIdentifier inconsistent with complete header
-    EXPECT_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
+
+    register_alias_type_object();
+    TypeIdentifierPair alias_type_ids;
```

Comment:
Same suggestion applies here. I think we can save resources reusing variables that are no longer being used.

Reply:
reused for all cases in [d3245e1](https://github.com/eProsima/Fast-DDS/pull/4496/commits/d3245e16dd687c62ce397be8018c718828d2bc6e)